# Measuring the brightness temperature of the sun using a DSTV dish
Last update: 17/09/2026


### DARA AVN SCHOOL (SARAO/HartRAO)
Instructors: Dr Pfesi van Zyl

<img src="https://www.spacetv.co.za/wp-content/uploads/2018/11/D90-O-A-T-SpaceHD-90cm-Aluminium-Dish-DStv-Approved.jpg" width="300" height="300">

This notebook turns a small Ku-band satellite dish into a simple radio telescope. You will calibrate the radiometer, estimate the dish beam size, measure the angular size of the Sun, and use beam dilution to infer the Sun's **brightness temperature at about 12 GHz**.

## Learning outcomes

By the end of the tutorial you should be able to:

1. Explain what a radio telescope actually measures.
2. Distinguish brightness temperature, antenna temperature, and system temperature.
3. Estimate the HPBW of a circular dish.
4. Calculate beam and source solid angles.
5. Calibrate arbitrary detector units into kelvins using hot and cold loads.
6. Correct for beam dilution.
7. Estimate the Sun's radio brightness temperature.
8. Propagate measurement uncertainties.
9. Explain why the radio Sun is hotter than the optical photosphere.

> **Solar safety:** Never look directly at the Sun through optics. The pinhole method uses projection only.


***KEY QUESTION***: 
What does the number coming out of a radio telescope actually mean? The
entire tutorial is an answer to this question.

## Index

1. Introduction

- 1.1. theory

- 1.1.1 Radio waves
- 1.1.3 Brightness Temperature and Antenna Temperature 
- 1.1.4 Angular Sizes and the Sun’s Brightness Temperature

- 1.2. Detecting Radio Emission from Space

2. The Experiment

- 2.1. Equipment

- 2.1.1 A standard small Ku-band (12GHz) satellite dish.
- 2.1.2 Apparatus for Measuring the Diameter of the Sun
- 2.1.3 Apparatus for Measuring the Antenna Beamwidth

- 2.2 Experimental procedure

- 2.2.1 Calibrating the Radio Telescope
- 2.2.2 Measure the Antenna Temperature from the Sun
- 2.2.3 Measure the Angular Diameter of the telescope beam

- 2.2.4 Measure the Angular diameter of the Sun
    - 1. Estimating the temperature of the Sun using the beam size
    - 2. Estimating the solar radius of the Sun using the angular diameter
    - 3. Estimating the angle subtended by the Sun using the solar radius
    - 4. Find ratio of the areas of the beams
- 2.2.5 Calculate the Brightness Temperature of the Sun


# 1. Introduction

<!-- A satellite dish can be used as a remote sensing device for measuring temperature. This relies on the fact that the measured radio power emitted by an object is proportional to its temperature (i.e. technically, the radio emission from most objects occurs in the Rayleigh-Jeans tail of the Black Body brightness distribution).  In particular, a satellite dish can be used to measure the brightness temperature of the Sun at the dish operating frequency. -->

## 1.1 Theory

### 1.1.1 Radio waves

##### Radio waves -> electromagnetic signals having wavelengths greater than a centimetre (see Fig 1. below). 

E.g. 
- Commercial **FM radio** operates in the frequency range from **88 to 108 MHz, (λ ~ 3 metres)**. 
- **Cellphones** operate at a frequency $\nu$ of about **900 MHz, (λ ~ 33 cm)**.



<!-- <img src="VisibleLightSpectrum.jpg" alt="VisibleLightSpectrum.png" width="800" height="200"> -->
<img src="imgs/em2.jpg" alt="em2.jpg" width="800" height="200">

Fig 1. credit: https://i0.wp.com/apniphysics.com/wp-content/uploads/2023/10/electromagnetic-spectrum-wavelengths.jpg?ssl=1

### The microwave band is the short wavelength part of the radio band and covers 1 to 30 cm wavelength.

- **Microwave ovens** operate at **2.4 GHz (λ ~ 12 cm)**. 

- **DSTV satellites** transmit at **12 GHz (λ ~ 2.5 cm**). 



In [13]:
c_light = 2.99792458e8      # m/s
nu = 12e9                  # Hz
wav = c_light / nu        # wavelength 

print(f"Wavelength at 12 GHz = {wav:.4f} m = {wav*100:.2f} cm")


Wavelength at 12 GHz = 0.0250 m = 2.50 cm


> ### Why are cellphones are not allowed at a radio observatory ?

With the advent of technology comes great consequences

<img src="imgs/wavesforshutterstock-01.jpg" alt="EM_range2 plot" width="800" height="200">

Fig 2. credit: https://spectraldesign.com.au/wp-content/uploads/2023/02/wavesforshutterstock-01.jpg

Everything between 1 and 30cm makes radio observations impossible.



<img src="freq.png" alt="freq.png" width="800" height="200">
 For example: Your favourite mobile phone bands

<img src="imgs/ska.png" alt="ska.png" width="800" height="200">

Fig 3. SKA-mid operating freqs. **This will be the most sensitive radio telescope in the world.**
<!-- # <img src="em2.jpg" alt="em2.jpg" width="800" height="200"> -->

### 1.1.2 Radio Telescope Antennas

> **radio telescope** focuses EM waves with a dish \and measures their power with a **radiometer**.



<img src="imgs/radio-telescope.webp" alt="radio telesope" width="600" height="200">

Fig 4. Anatomy of a radio telesope.

A “classic” radio telescope for use in the microwave band has a circular parabolic reflector with a feed horn
at the focus to collect the incoming microwaves and pass them to  amplifiers in the receiver. 

> **A DSTV satellite dish also works in this way.**

- can be used as a **mini-radio telescope by replacing the DSTV decoder with a radiometer (in our case an [LNA](https://en.wikipedia.org/wiki/Low-noise_amplifier) (low noise amplifier)) for measuring the signal strength**.



#### 1.1.2.1 What number comes out of a radio telescope ?

From the Sun to a number on the meter, what happens to the radio signal ? (see Fig 5.)


A radio telescope does not "measure temperature" directly. It measures power. Calibration lets us express changes in that power as an equivalent temperature. **(i.e. we measure changes in the electronics that occur when you are looking/pointed at a source)**


<span style="color:blue">**Sky → Dish → Feed / LNA → Detector / Radiometer → Meter → Calibration → Antenna temperature**</span>


☀ Sun / Sky  
↓  
📡 Dish collects the radio waves  
↓  
Feed / LNA receives and amplifies the weak RF signal with minimal added noise  
↓  
Detector / Radiometer measures the received power  
↓  
Meter displays or gives the radiometer output as a number in arbitrary units  
↓  
Calibration converts that number into Kelvins or temperature

<img src="imgs/signalchain.png" alt="signal chain" width="800" height="200">

Fig 5. Radio emission signal chain 

 #### 1.1.2.2. How does the telescope respond to radio waves (resolution) ?

To understand how a **reflector antenna responds to radiation** coming from different angles, consider what
happens when a plane wave of wavelength λ arrives at a circular aperture of diameter D (Fig. 6). 



<img src="imgs/images.png" alt="youngs slit experiment" width="800" height="200">

<!-- <img src="images.png" align="left" width="45%">
<img src="images.png" align="left" width="45%">
 -->
Fig 6. Youngs slit experiment. 

**Constructive and destructive interference produces a circularly symmetric diffraction** pattern, with a central
maximum and concentric rings of decreasing strength (Fig 7 (a) and (b)).


<!-- <img src="images.png" alt="youngs slit experiment" width="800" height="200"> -->

<!-- <img src="diffraction2.png" align="left" width="45%">
<img src="diffraction1.png" align="left" width="45%">
 -->


<img src="imgs/focus.png" alt="focus.png" width="800" height="200">

- An “ideal” antenna would produce a beam that captures 100% of the incoming energy in the main beam and would have no sidelobes. 


- => “main beam efficiency” $\epsilon_{m}$ of 1.0. 


- This describes the diffraction pattern where nothing obstructs the path of the waves, i.e. it has an “**unblocked** aperture”, and is **uniformly illuminated**. 


- In practice, not possible to actually achieve this, and **$\epsilon_{m}$ usually lies between 0.6 and 0.8**.

Fig 7 (c) and (d) show an ideal and an actual beam pattern in cross-section on linear and logarithmic scales.

**The “ideal” pattern has been modelled here with a parabolic shape, while the mathematical form of the real
pattern is a sinc$^{2} or \bigg(\frac{sin (x)}{x}\bigg)^{2}$ function.** 




This same pattern describes the **response of a circular antenna to plane waves** coming from different angles,  called an <span style="color:red">**antenna beam pattern**</span>. (see Fig 8, left)
A circular aperture does not respond only to one exact direction. It has a main lobe and sidelobes. 

<img src="imgs/beams.png" alt="beam pattern" width="800" height="200">

Fig 8. What the telescope sees is radio emission at different strengths in a particular direction in the sky. For a real dish, feed illumination, blockage, shape errors and the offset geometry
modify the exact pattern, see Fig 8 (b).

<img src="imgs/beamz.png" alt="beam pattern" width="800" height="200">

Fig 9. What the telescope sees is radio emission at different strengths in a particular direction in the sky. 


<!-- - The **first minimum or null in the pattern occurs at a radius of about  1.2$\frac{λ}{D}$ radians**, so the beam width to the first nulls is
 -->

<span style="color:red">**Antenna beam / HPBW**</span>: resolution - angular size of the main lobe of the beam at the points where the power drops to half of the maximum value.
- The beamwidth at the half-power points (HPBW), also called the Full Width at Half Maximum (FWHM), is about half this, as shown in fig 9.

$\begin{equation} 
\label{eq:hpbw} \tag{2}
 HPBW = FWHM = k\frac{\lambda}{D} [radians]
\end{equation}$


For a circular dish, the scaling factor k depends on illumination. An ideal uniformly illuminated aperture has an
Airy-pattern FWHM near 1.03 λ/D. Real radio dishes are often approximated with k 1.1-1.2.
For this exercise we use 1.2 unless you measure the beam directly.


<span style="color:green">
$\begin{equation} 
\label{eq:hpbw} \tag{2}
 HPBW = 1.2\frac{\lambda}{D} [radians]
\end{equation}$
    </span>

For the 26m dish at HartRAO at 12 GHz, where D = 26 m and $\lambda$ = 2.5 cm = 0.025 m

$\begin{equation} 
\label{eq:hpbw} \tag{2}
 HPBW = 1.2\frac{0.025}{26} = 1.154e-3 \qquad rad
\end{equation}$



converting to degrees

$\begin{equation} 
\label{eq:hpbw} \tag{3}
 HPBW = 1.154e-3 * 57.3 = 0.066 \qquad deg
\end{equation}$


or

$\begin{equation} 
\label{eq:hpbw} \tag{4}
 HPBW = 4 \qquad arcmin
\end{equation}$



So at 12 GHz, the 26 m telescope's main beam is roughly 0.1 degrees or 4 arcminutes wide at half power. HPBW tells you the width of the sky your telescope is most sensitive. i.e. When I point my telescope somewhere, the strongest part of its sensitivity covers roughly a 0.1 degree-wide region of the sky.



<img src="imgs/beam_hart.png" alt="hartrao beam at 12 GHz.png" width="800" height="200">

Fig 10. Telescope beam pattern

For the frequencies monitored at HartRAO


<img src="imgs/beam_comparison_hart.png" alt="beam_comparison_hart" width="800" height="200">

Fig 11

<!-- For comparison, The Earth rotates 1 degree in 4 minutes. 

360° in 24 h = 15° per hour = 1° every 4 minutes.

Question: How long do you think it would take  -->

### 1.1.3 Brightness Temperature and Antenna Temperature 

#### 1.1.3.1 Brightness temp

Brightness temperature is a way of saying:

<span style="color:blue">**“How bright does this object look in radio waves, expressed as a temperature?”**</span>

At radio wavelengths, hotter-looking radio sources give stronger signals. So astronomers convert the measured radio brightness into a temperature scale measured in kelvin. We call this brightness temperature, $T_{B}$.

***"The apparent temperature of an emitting source at a given frequency"***

Fig. 12 shows the brightness as a function of frequency for several black body radiators modelled as having
equal size but different temperatures. 

- hotter objects produce more radiation than cooler ones, and the brightness maximum occurs at a higher frequency. 

- objects with temperatures more than a few degrees above absolute zero, the brightness peak occurs well above the operating range of radio telescopes. 

<img src="imgs/blackbody.png" alt="blackbody.png" width="500" height="200">

Fig 12. Blackbody spectrum

Radio telescopes work in the range where h$\nu$ << $kT$ , 

- the Planck function approaches the Rayleigh-Jeans form when h$\nu$ << $kT$ 
- In this limit, specific intensity is proportional to temperature. 

From Rayleigh-Jeans law we are able to express the brightness B of a radio emitting source using the temperature $T_{B}$ of the emitting source:

$\begin{equation} 
\label{eq:tb} \tag{5}
 B = \frac{2kT_{B} \nu^{2}}{c^{2}} 
\end{equation}$



where 

<span style="color:green">
$\begin{equation} 
\label{eq:tb} \tag{6}
 B = \frac{2kT_{B}}{\lambda^{2}} \quad [W m^{-2} Hz^{-1} sr^{-1}]
\end{equation}$
</span>


where $k$ = Boltzmann’s constant = 1.38 x 10$^{-23}$ [$J K^{-1}$].

## **There is also an important difference between antenna temperature and brightness temperature.**

#### 1.1.3.2 Antenna temp


<!-- 
For some astronomical objects the brightness temperature that is measured using a radio telescope is meaningful as a physical temperature, for example when observing a planet or moon. For other objects it may
not be, depending on the mechanism that produced the radio emission. -->
<!-- 
**By pointing the antenna at objects of known temperature that completely fill the beam, we can calibrate
the output signal in units of absolute temperature (Kelvins). So one can think of a radio telescope as a
remote-sensing thermometer.** -->

The “${\textbf{antenna temperature}}$” $T_{A}$ of a source is <span style="color:blue">**the radio power/emission expressed as the increase in temperature measured
when the antenna is pointed at a radio-emitting source.**</span>

- It will be less than brightness temperature of the source if the source does not fill the whole beam of the telescope. This is **beam dilution.**
    - Filled beam -> all emission
    - source beam < antena beam

- Note that “antenna temperature” has nothing to do with the physical temperature of the antenna.



<img src="imgs/sunbeam.png" alt="sunbeam" width="800" height="200">

Fig 13. 

----------------------------------
**--- So ---**


> **Brightness temperature $T_{B}$** = how bright the source itself is


> **Antenna temperature $T_{A}$** = how bright the source appears to this telescope

-----------------------------------

### 1.1.4 Angular Sizes and the Sun’s Brightness Temperature

To obtain the brightness temperature $T_{B}$ of the emitting source from its measured antenna temperature $T_{A}$,
we have to 

- measure the <span style="color:blue">**angular size (area) of the source**</span> and of the <span style="color:blue">**telescope beam (beam solid angle)**</span>. 

The ratio of the angular size (solid angle) of the source to the angular size of the beam gives the fraction of the beam that is filled by the source.

This will be useful to remember during the experiment later on.

The difference between physical diameter and angular diameter is shown in Fig. 14. For example, the Sun
has a physical diameter of 1.4 million km, while the Moon has a diameter of 3500 km. Yet, as seen from
the Earth, the Sun and the Moon appear to be the same size, 

- i.e. they have the same angular diameter.

How can this be? The Sun is 400 times bigger than the Moon, but it is also 400 times further away. 

As the
projection of the antenna beam onto the sky is two-dimensional, we shall need to find the angular area that
it covers. 

**Angular area is called a “solid angle” and the units are radians$^{2}$, or steradians (sr).**

An object
with an angular radius θ radians subtends a solid angle

<img src="imgs/sunmoon.png" alt="sun and moon angular size.png" width="800" height="200">

Fig 14.

<img src="imgs/radii.png" alt="radii.png" width="800" height="200">

$\begin{equation} 
\label{eq:4} \tag{4}
 \Omega_{s} = 2 \pi (1 - cos \theta) \quad [sr]
\end{equation}$

For small $\theta$, i.e. where $\theta$ << 1,  the angular size is given as 

<span style="color:green">
$\begin{equation} 
\label{eq:5} \tag{5}
 \Omega_{s} = \pi \theta^{2} \quad [sr]
\end{equation}$
  </span>

We can use this equation to calculate the solid angle of the Sun as seen from the Earth, $\Omega_{s}$.

The telescope beam  $\Omega_{A}$ of the antenna can be obtained from the half-power beamwidth (HPBW) - in units
of radians - by assuming the main lobe of the beam has a Gaussian shape:

$\begin{equation} 
\label{eq:6} \tag{6}
 \Omega_{A} \sim 1.333 (HPBW)^{2} \quad [sr]
\end{equation}$

For small $\theta$, the brightness temperature of the source can then be given as

<span style="color:green">
$\begin{equation} 
\label{eq:7} \tag{7}
 T_{B} = \frac{\Omega_{A}T_{A}}{\Omega_{s}\epsilon_{m}} \quad [K]
\end{equation}$
</span>

## 1.2. Detecting Radio Emission from Space

When the telescope looks at a radio source in the sky, the receiver output is the sum of the radio waves
received from several different sources:


- Behind the radio source whose brightness we want to measure is the cosmic microwave background (CMB) coming from every direction in space. This is the radiation emitted as the first atoms formed, 380 000 years after the Big Bang. The black body temperature of the CMB Tcmb has now decreased to 2.7 Kelvins, as the expansion of the Universe has stretched out the waves by a factor of 1000. The CMB produces a brightness temperature $T_{Bcmb}$ of $\sim$2.7 K at 1.4 or 4 GHz, reducing to 2.5 K at 12 GHz.



- The emission from the radio source we want to measure, which produces the antenna temperature $T_{A}$.


- Radiation from the dry atmosphere $T_{at}$ and from the water vapour in the atmosphere $T_{wv}$. The dry air adds about 1K, and at 12 GHz water vapour adds 1 - 2 K, depending on the humidity.

- The radiation the feed receives through the antenna sidelobes from the (warm - 290 K) ground or nearby buildings beyond the edge of the antenna, of brightness temperature $T_{g}$. With the antenna pointing straight up at zenith this could add 5 - 15 K; 10 K is a reasonable number to use. It increases when pointing close to the horizon.

- The amplifiers in the receiver generate their own electronic noise and so produce a receiver noise temperature $T_{R}$.

The sum of these parts is called the “system temperature” $T_{sys}$. Summing from the most distant noise
contributor to the nearest we have:

<span style="color:green">
$\begin{equation} 
\label{eq:8} \tag{8}
 T_{sys} = T_{Bcmb} + T_{A} + T_{at} + T_{wv} + T_{g} + T_{R} \quad [K]
\end{equation}$
</span>

For the Brightness temperature , we need to find $T_{A}$ from the Tsys measured by our detector

# 2. The Experiments

**In this experiment we are going to determine the brightness temperature of the sun by hand to give you hands on experience on how this can be done with very little equipment.**

The aim is to estimate 


<span style="color:magenta">
$\begin{equation} 
\label{eq:7} \tag{7}
 T_{B} = \frac{\Omega_{A}T_{A}}{\Omega_{s}\epsilon_{m}} \quad [K]
\end{equation}$
</span>


To do this, we need to find:

1. The antenna/telescope beam  $\Omega_{A}$

2. The angular size of the sun $\Omega_{s}$

3. The antenna temperature $T_{A}$

4. The temperture brightness $T_{B}$

We determine 1 - 3 in 3 steps and use those results to calculate 4.

1. We calibrate our equipment to get a conversion factor (c) [DSTV dish Hot/Cold load experiment]

2. Measure the temperature of the sun. [DSTV dish Hot/Cold load experiment]

3. Measure the angular diameter of the sun [Pinhole experiment]


-----------------------

You are now going to use the principles discussed in the previous section to calculate the brightness temperature of the Sun. But before you start, we need to setup our experiment. Here's what we are going to need:

### 2.1 Equipment

#### 2.1.1. A standard small Ku-band (12GHz) satellite dish. 

The main parts of a simple radio telescope comprising a satellite dish and radiometer are shown in the figure below. 

The dish output is fed to a radiometer. A “radiometer” means a “radio-meter” –
i.e. a device that measures the strength of the radio signal coming from the receiver on the dish.
Since the incoming radio waves from natural emitters are weak and noise-like. If the output of the detector is
connected to a loudspeaker, the signal sounds like a hiss, as one hears if a radio is tuned off-station. The
internal noise produced in the amplifiers is generally larger than the signal from natural radio sources.


<img src="imgs/antenna2.png" alt="antenna2.png" >

#### 2.1.2. Apparatus for Measuring the Diameter of the Sun

You will need: 
 - two pieces of card or paper, 
 - a pencil, 
 - a sharp knife, 
 - a ruler and 
 - a 2 metre tape measure.

#### 2.1.3. Apparatus for Measuring the Antenna Beamwidth

You will need:
- pen and 
- notepad or paper.


- (Optionally): Apparatus required is a tripod or mount on which the satellite dish can be locked in
position, a timer (e.g. watch on which seconds are displayed). We wont be using this for our experiment.


### 2.2 Experimental procedure


The temperature brightness $T_{B}$ is described in eq. (7) as 


$\begin{equation} 
 T_{B} = \frac{\Omega_{A}T_{A}}{\Omega_{s}\epsilon_{m}} \quad [K]
\end{equation}$

where 

- $\Omega_{A}$ is the beam solid angle, 
- $\Omega_{s}$ is the solid angle of the Sun as seen from Earth,    
- $\epsilon_{m}$ is the main beam efficiency and 
- $T_{A}$ is the emission from the radio source we want to measure. 

In order to measure $T_{A}$ we need to establish a conversion factor or "Calibrate our telescope" so that we can be able to express our detected radio emission it in terms of temperature in Kelvins [K].



#### 2.2.1 Calibrating the Radio Telescope

When a radio telescope detects radio emission from a celestial object, the dish produces an output voltage proportional to the object's radio power (a "square-law" detector) plus its own internal receiver noise. This voltage is displayed on a meter with an arbitrary scale. To get the temperature brightness of the object, we need a way to establish a scale of Kelvins per radiometer output unit. To establish a scale of Kelvins per radiometer output unit, we use a technique that leverages the sky and ground as reference points. 

##### Determining the Scaling Factor

We can determine this scaling factor by using the sky as a "cold load" and the ground as a "hot load."

- **Cold Load (Sky)**: By pointing the telescope at the zenith, we effectively observe a "cold load" with a temperature $T_{sky} \sim$ 10 K. This temperature accounts for the Cosmic Microwave Background, atmospheric contributions, and ground radiation.

- **Hot Load (Ground)**: Aiming the telescope at the ground provides a "hot load" with a temperature $T_{g} \sim$ 300K on a warm day.
 

Let:
 - $V_{1}$:Meter reading for the sky
 - $V_{2}$: Meter reading for the ground
 - $T_{R}$: Receiver noise temperature
 - $T_{sky}$: Effective sky temperature
 - $T_{g}$: Ground temperature
- c: Constant of proportionality

if $V_{1}$ and $V_{2}$ are the meter readings for the sky and ground, respectively. The relationship between temperature and voltage can be expressed as:
  
$\begin{equation} 
\label{eq:9} \tag{9}
 T_{R} + T_{sky}= cV_{1} \quad [K]
\end{equation}$ 

where $T_{sky} = T_{Bcmb} + T_{at} + T_{wv} + T_{g} \sim 10 K.$

$\begin{equation} 
\label{eq:10} \tag{10}
 T_{R} + T_{g}= cV_{2} \quad [K]
\end{equation}$ 

By solving these equations simultaneously, we can determine the value of $c$ and establish a calibrated temperature scale for the radiometer.


##### HartRAO Satellite dish Radiometer


On the HartRAO satellite dish radiometer, typical meter readings are:

10: Aimed at the sky ($V_{1}$)

30: Aimed at the ground ($V_{2}$)

24: Aimed at the Sun ($V_{3}$), you will need this for the next calculation

By using these readings and the known temperatures of the sky and ground, we can calculate the scaling factor c and subsequently determine the temperature of the Sun (or any other celestial object) based on its corresponding meter reading.

##### >>> Solution for Determining the Scaling Factor

Step 1: Establish the equations:

As given in the experiment notes, we have two equations:
  
$\begin{equation} 
\label{eq:9} \tag{9}
 T_{R} + T_{sky}= cV_{1} \quad [K]
\end{equation}$ 

$\begin{equation} 
\label{eq:10} \tag{10}
 T_{R} + T_{ground}= cV_{2} \quad [K]
\end{equation}$ 
 
Step 2: Solve for the scaling factor, c:

To find the scaling factor, we can subtract equation 9 from equation 10:

  
$\begin{equation} 
 (T_{R} + T_{g}) - (T_{R} + T_{sky}) = cV_{2} - cV_{1} 
\end{equation}$ 

This simplifies to:

$\begin{equation} 
 (T_{g} - T_{sky}) = c(V_{2} - V_{1}) 
\end{equation}$ 

Therefore, the scaling factor, c, is:

$\begin{equation} 
 c=\frac{T_{g} - T_{sky}}{V_{2} - V_{1}}
\end{equation}$ 	
 

Step 3: Calculate the scaling factor using given values:

$\begin{equation} 
 c=\frac{300 K - 10 K}{30 - 10} = \frac{290 K}{20} = 14.5 \quad K/unit
\end{equation}$ 



In [2]:
Tg=300
Tsky=10
V1=10
V2=30

c= (Tg-Tsky)/(V2-V1)

f'The scaling factor is {c} kelvins per arbitrary unit'

'The scaling factor is 14.5 kelvins per arbitrary unit'

Now that we know $c$ we can determine the noise contribution from $T_{R}$

substituting for $c$ in equation (9)

$\begin{equation}
 14.5(10) = T_{R} + 10
\end{equation}$ 

$\begin{equation}
 T_{R} = 145 - 10
\end{equation}$ 

$\begin{equation}
 T_{R} = 135 \quad [K]
\end{equation}$ 


In [3]:
TR = int((c*V1)-Tsky)

f'The noise temperature of the reciever is {TR} K'

'The noise temperature of the reciever is 135 K'

#### 2.2.2 Measure the Antenna Temperature from the Sun

We need to aim the telescope at the Sun. With an offset-fed paraboloid this can be a litle tricky. Hold
the dish horizontal. then turn the dish horizontally so the shadow of the feed arm falls across the centre
of the dish, demonstarated her by Mike in the image below. 

<img src="imgs/mike.png" alt="mike.png" >

Then rotate the dish in elevation so the shadow of the feed arm on the dish gets shorter. The
Sun comes into the beam just after the shadow leaves the edge of the dish. Adjust the direction gently up
and down and sideways to maximise the signal from the Sun; the signal should roughly double. Once the telescope is accurately pointed at the Sun, record the new meter reading, $V_{3}$. This reading corresponds to the combined temperature of the receiver, sky, and Sun, and is expressed as:

$\begin{equation} 
\label{eq:11} \tag{11}
cV_{3} =  T_{R} + T_{sky} + T_{Asun} 
\end{equation}$ 

Calculating the Sun's Antenna Temperature:

Since the scaling factor, $c$, and the receiver temperature, $T_{R}$, are known from the calibration process, and we have a reasonable estimate of the sky temperature, $T_{sky}$, we can use equation 11 to directly calculate the antenna temperature of the Sun, $T_{Asun}$.

$\begin{equation} 
T_{Asun} = cV_{3} -  T_{R} - T_{sky}  
\end{equation}$ 


#### >>> Solution for Measuring the Antenna Temperature from the Sun

$\begin{equation} 
\label{eq:17} \tag{12}
T_{Asun}  = (14.5 K/unit)(24 units) - 135 K - 10 K 
\end{equation}$ 

$\begin{equation} 
T_{Asun}  = 348 - 145 K 
\end{equation}$ 

$T_{Asun}$ = 203 K

 

In [14]:
V3=24

TAsun = int(c*V3) - TR - Tsky

f'The antenna temperature of the sun is {TAsun} K'

'The antenna temperature of the sun is 203 K'

But the Sun is much hotter than this! To understand why this is so, and to work out the real
temperature of the Sun, we must consider the telescope's field of view. We have to figure out how much of the sky the dish “sees” (or more technically, the beam solid angle) at a given time.


#### 2.2.3 Measure the Angular Diameter of the telescope beam 

<!-- 
Next we need to measure the angular diameter of the Sun and of the telescope beam in order to calculate
the solid angles they subtend. -->

<!--img src="FOV.jpg" alt="FOV.jpg" -->

<!-- To accurately calculate the solid angles subtended by the Sun and the telescope's beam, we must first measure their respective angular diameters. -->

##### 1. Angular diameter or angular resolution of The telescope beam

A radio telescope, like the satellite dish you're using, is sensitive to radio waves originating from a specific region of the sky. This region is known as the antenna's "beam."

##### Understanding Beamwidth
What is the diameter of this beam ? 

As the radio source moves away from the middle of the beam, the angle of the waves hitting the dish changes. When all the waves coming from each part of the dish are in phase, we get the strongest signal, when the
emitter is in the centre of the beam. Moving away from the centre, the waves from different parts
of the dish start to become out of phase with each other, causing destructive interference. The
telescope sensitivity falls to a minimum when the incoming wave is tilted at an angle so that
there is a phase difference of approximately one wavelength across the diameter of the dish.
This minimum is called the “first null” in the antenna beam pattern. We normally use the points
where the response of the antenna has fallen to half that in the centre of the beam defined as the
“half-power beamwidth” ($\theta_{HPBW}$) or angular resolution.

$\theta_{HPBW}$ ~ wavelength / diameter of the dish, in units of radians (1 radian = 57.3 degrees). 

For a circular aperture, like our dish, the angular resolution, $\theta_{HPBW}$, can be estimated using the Rayleigh criterion given in equation (2).

* Calculate the HPBW of the telescope beam.

${\textbf{For comparison, also calculate the resolving power of your eye (2 mm pupil) 
and of a 20cm optical telescope, using}}$
${\textbf{ a wavelength of 500 nm)}}$.

##### HartRAO Satellite dish parameters

dish_diameter = 50 # cm, satellite dish diameter

wavLen = 2.5 # cm, observing wavelength

#### >>> Solution for measuring the Angular diameter or angular resolution of The telescope beam

1. Calculation of the $\theta_{HPBW}$

$\begin{equation} 
\label{eq:13} 
\theta_{HPBW} =  (1.2)(obs\_freq) / dish\_diameter
\end{equation}$

$\begin{equation}
\theta_{HPBW} =  (1.2)(2.5) / 50
\end{equation}$

$\begin{equation} 
\theta_{HPBW} =  0.060 \quad [radians]
\end{equation}$

= 3.4 degrees (approx)


In [15]:
import numpy as np

dish_diameter = 50 
wavLen = 2.5
theta_hpwb_rad = (1.2*wavLen)/dish_diameter
theta_hpwb = np.rad2deg(theta_hpwb_rad) # convert radians to degrees

f'The hpbw or angular diameter of the antenna beam is {theta_hpwb_rad:.2f} radians or ~{theta_hpwb:.2f} degrees'

'The hpbw or angular diameter of the antenna beam is 0.06 radians or ~3.44 degrees'

This is the beam of the telescope. We now need to figure out what is the beam solid angle $\Omega_{A}$ or the angle subtended in the sky by the telescope beam. This is a crucial parameter in radio astronomy as it determines the amount of celestial radiation that the telescope can collect or more technically a measure of the telescope's field of view on the sky. A larger beam solid angle means the telescope collects radiation from a larger area of the sky, while a smaller beam solid angle provides a more focused view. 

Calculate the antenna beam solid angle $\Omega_{A}$

#### >>> Solution for measuring the solid angle or area of The telescope beam

1. Calculation of the solid angle $\Omega_{A}$

The beam area (solid angle) is agiven in eqn. (6)

$\begin{equation} 
\Omega_{A} =  1.133 (theta\_hpbw)^{2} \quad [sr]
\end{equation}$

$\begin{equation} 
\Omega_{A} =  1.133 (3.44)^{2} \quad [sr]
\end{equation}$

= 13.39

In [16]:
import math 

omega_a_deg = 1.133 * (theta_hpwb)**2

f'The solid angle or area of the antenna beam is {omega_a_deg:.2f} square degrees'

'The solid angle or area of the antenna beam is 13.39 square degrees'

#### 2.2.4 Measure the Angular diameter of the Sun

To measure the angular diameter of the Sun, we will use the pinhole projection experiment. Assuming that our target, the Sun, is much smaller or can fit within the beam of our antenna, we now want to find out what its 
angular diameter is.

Put simply, we need to know how many times the Sun would fit into the beam of the telescope.

##### 1. Estimating the temperature of the Sun using the beam size
<img src="imgs/pinhole1.png" alt="pinhole1.png">  



Use pinhole projection to measure the angular diameter of the Sun. On one piece of card use a ruler
and pencil to mark three equilateral triangles several cm apart, with sides of about two, three and four
millimetres. Cut out the triangles using the sharp knife and ruler. The card with the holes is used to project images of the Sun onto the second card (Fig above and below). The two cards
need to be separated by a distance D, measured by the tape measure. 

**If the dish can be pointed accurately using a tripod, the beamwidth could actually be measured,
by scanning across the Sun or letting it drift through the beam. This is one of the methods used to
observe radio sources by the HartRAO 26m telescope.**

<img src="imgs/pinhole2.png" alt="pinhole2.png"  >

##### HartRAO Satellite dish parameters

D = 2000: mm, length of distance between the 2 papers or cards (from 2, above) measured with a tape measure

d = 20: mm, The linear diameter of the projected Sun measured with the ruler.


#### >>> Solution for measuring the Angular diameter or angular resolution of The telescope beam

The angular diameter of the Sun is d/D radians:

$\begin{equation} 
\theta_{sun} =  \frac{d}{D} \quad [radians]
\end{equation}$

$\begin{equation} 
\theta_{sun} =  \frac{20}{2000} \quad [radians]
\end{equation}$

$\begin{equation} 
\theta_{sun} =  0.01 \quad [radians]
\end{equation}$

or 
 
 $\begin{equation} 
\theta_{sun} =  0.57^{\circ} 
\end{equation}$

In [7]:
d=20
D=2000
theta_sun = d/D
theta_sun_deg = np.rad2deg(theta_sun)
f'The angular diameter of the Sun = {theta_sun} [radians] or {theta_sun_deg:.2} degrees'

'The angular diameter of the Sun = 0.01 [radians] or 0.57 degrees'

##### 2. Estimating the solar radius of the Sun using the angular diameter

The solar radius is given as 

$\begin{equation} 
\theta_{s} =  \frac{d}{2D} \quad [radians]
\end{equation}$

$\begin{equation} 
\theta_{s} =  \frac{20}{2*2000} \quad [radians]
\end{equation}$


In [8]:
d=20
D=2000
theta_s = d/(2*D)
theta_s_deg = np.rad2deg(theta_s)
f'The solar radius of the Sun = {theta_s} [radians] or {theta_s_deg:.2} degrees'

'The solar radius of the Sun = 0.005 [radians] or 0.29 degrees'

##### 3. Estimating the angle subtended by the Sun using the solar radius

The angle subtended by the sun or are of the sun is given by equation 5

$\begin{equation} 
\Omega_{s} =  \pi\theta^{2} \quad [sr]
\end{equation}$

$\begin{equation} 
\Omega_{s} =  \pi(theta\_s\_deg)^{2} \quad 
\end{equation}$

$\begin{equation} 
\Omega_{s} =  \pi(0.29)^{2} \quad 
\end{equation}$

$\begin{equation} 
\Omega_{s} =  0.26 \quad [degrees \quad squared]
\end{equation}$

**Note** If you perform this calculation using the radian results from the previous calculation, your units will be radians squared for the answer. Since we used degress, our answer is stated in squared degrees. 

In [9]:

omega_s_deg = math.pi*(theta_s_deg)**2

f'The angle subtended by the Sun = {omega_s_deg:.2} degrees squared'

'The angle subtended by the Sun = 0.26 degrees squared'

##### 4. Find ratio of the areas of the beams

In [10]:


area_ratios = int(omega_a_deg/omega_s_deg)

f' and so put simply we could fit {area_ratios} Suns into the beam of the dish.'


' and so put simply we could fit 51 Suns into the beam of the dish.'

#### 2.2.5 Calculate the Brightness Temperature of the Sun

To calculate the brightness temperature of the Sun we now have the Sun’s solid angle $\Omega_{s}$ from Eqn. 5, the
beam solid angle $\Omega_{A}$ from Eqn. 6, and the antenna temperature measured from the Sun $T_{A}$, from Eqn. 11.

However, we need to correct the antenna temeperature to reflect the area ratio calculated above


In [11]:
TAsun_corrected=area_ratios*TAsun
f'Hence the temperature of the sun is about {TAsun_corrected} K'

'Hence the temperature of the sun is about 10353 K'

This does not allow for the efficiency of the dish. Experiments indicate that a
reasonable value for ǫm for a DSTV dish at 12 GHz is about 0.75. For mesh surface satellite dishes working
at 3.8 GHz, a lower value is more likely as the surface shape is less accurate, and ǫm is about 0.5. For a
mesh dish with a 1.4GHz can feed, ǫm is about 0.5. These values are based on measurements with domestic
satellite dishes. Correcting for this, the Sun's brightness temperature is about 

In [12]:
Tb=int(TAsun_corrected/0.75)
f'So the brightness temperature of the sun can be estimated as {Tb} Kelvins at 12 GHz'

'So the brightness temperature of the sun can be estimated as 13804 Kelvins at 12 GHz'

Estimate the uncertainty in your result in the usual way, by propagating estimates of the error in each of
your measurements. Assume an uncertainty in $\epsilon_{m}$ of 10%.

This temperature will vary during the 11 year solar cycle, but it reasonably matches the 
value given in standard references for this observing wavelength. However, the T$_{B}$ is about 
three times higher than the value of 5800 K measured at visible wavelengths.

How does your result for the Sun’s brightness temperature compare to the temperature usually quoted for
the Sun’s photosphere (light emitting surface)? What do you think this implies?


In essence, we cannot treat the Sun as a perfect blackbody radiator – some other mechanism must be 
increasing the emission from the Sun at radio wavelengths. What could this be?

<!-- <img src="pinhole3.png" alt="pinhole3.png"  width="500" height="200" > -->